### ISSUE TITLE
[NEOS-1630] Optimize Subset table in same fashion as job mappings table

### ISSUE THREAD

**Issue by nickzelei** [nucleuscloud/neosync#2967]
This table is much simpler but still suffers from the same performance issues that the job mappings table does.

It should be optimized in a similar fashion with memoization as well as pulling out the form components.

<sub>From [SyncLinear.com](https://synclinear.com) | [NEOS-1630](https://linear.app/nucleus-cloud/issue/NEOS-1630/optimize-subset-table-in-same-fashion-as-job-mappings-table)</sub>

**Comment by jasondamour** [comment]
Also would be fantastic if we could do the same bulk modification

**Comment by nickzelei** [comment]
> Also would be fantastic if we could do the same bulk modification

@jasondamour - what do you mean by this? Would the idea be here to bulk set the where clauses for multiple tables?

**Comment by jasondamour** [comment]
Yup. I.e. if all tables have a `tenant_id` column, and I would like to globally subset by one tenant, the ability to add where clauses across multiple tables would be nice

**Comment by nickzelei** [comment]
> Yup. I.e. if all tables have a `tenant_id` column, and I would like to globally subset by one tenant, the ability to add where clauses across multiple tables would be nice

@jasondamour - I just shipped this feature (https://github.com/nucleuscloud/neosync/pull/3101)
It is available on main and will be in next tagged release (v0.5.2)

**Comment by jasondamour** [comment]
Wow, TY!
